Pipeline/CreateUCDComponentVersion/README.md

```diff
@@ -1,69 +1,243 @@
-# IBM UCD Packaging based on IBM DBB Build Report
+# IBM UCD Packaging Based on IBM DBB Build Report
 
-An important step in the pipeline is to generate a deployable package. This sample groovy script
-- extracts information about the build outputs from the DBB BuildReport.json
-- generates the UCD shiplist ```shiplist.xml``` file
-- invokes the ```buztool.sh``` of UCD
+## Summary
+
+An important step in the pipeline is to generate a deployable package. This sample Groovy script:
+
+- Extracts information about the build outputs from the Dependency Based Build (DBB) `BuildReport.json`. The script is able to take a single DBB build report or multiple build reports to build a cumulative package across multiple incremental builds. 
+- Generates the UrbanCode Deploy (UCD) shiplist `shiplist.xml` file.
+- Invokes the `buztool.sh` with the appropriate configuration to store the binary package in the artifact repository and to register a new UCD component version.
+
+## High-level Processing Flow
+
+This section provides a more detailed explanation of how the CreateUCDComponentVersion script works and what it does.
+
+1. **Initialization**
+   1. Read command line parameters.
+   1. Read any application and global properties files that are passed as a comma-separated list via `--packagingPropFiles`.
+       (This parameter is optional for UCD package format v1, but required when using UCD package format v2 - see `--ucdV2PackageFormat`.)
+
+1. **Process the DBB build report(s)**
+   1. Either read DBB's `BuildReport.json` from the pipeline work directory, or loop through the list of provided DBB build reports (using the `--buildReportOrder` or `--buildReportOrderFile` option).
```

Comment:
Does that mean that, if you specify nothing (no parameter), the script will look for a file named BuildReport.json in the directory pointed by --workDir? The first part of the sentence would need some clarity, like "If either `--buildReportOrder` or `--buildReportOrderFile` parameter is specified, loop through the list of provided DBB build reports, or if not specified, read DBB's `BuildReport.json` file from the pipeline work directory pointed by the `--workDir` parameter"